extensions/amp-list/0.1/amp-list.js

```diff
@@ -1092,14 +1117,22 @@ export class AmpList extends AMP.BaseElement {
    * @private
    */
   attemptToFit_(target) {
-    if (this.element.getAttribute('layout') == Layout.CONTAINER) {
-      return;
-    }
     this.measureElement(() => {
       const targetHeight = target./*OK*/ scrollHeight;
       const height = this.element./*OK*/ offsetHeight;
       if (targetHeight > height) {
-        this.attemptChangeHeight(targetHeight).catch(() => {});
+        this.attemptChangeHeight(targetHeight).then(
+          () => {
+            if (!this.isLayoutContainer_) {
+              return;
+            }
+            setImportantStyles(this.element, {
+              'height': '',
+              'overflow': '',
+            });
+          },
+          () => {}
```

Comment:
Control flow in this class is confusing. :)

Do we intend to support layout=container for the load-more feature?

Reply:
I don't see a use case for it now, as `load-more="manual"` implicitly relies on a fixed height to manage its own resizing and `load-more="auto"` wouldn't behave differently with `layout=container` enabled. If there is a feature request for this later it may be worth revisiting the conversation, but I am not sure how valuable it is to undertake now.

Reply:
Cool, agreed. My underlying question was why do we "unlock height" here in `attemptToFit_` if most of its callers are skipped when `layout=container`.

I.e. if there's only a single code path where this happens, maybe just put the logic there and Promise chain it? 

```js
this.attemptToFit_(...).then(() => this.unlockHeight_());
```

Reply:
The `unlockHeight` is conditioned on a successful `attemptToChangeHeight`. In order to conditionally unlock in the single code path, we would have to bubble up any errors thrown by `attemptToChangeHeight` and catch them in all other code paths. IMO we should keep the unlock where it is now to avoid throwing unexpected errors in other places.

Reply:
> we would have to bubble up any errors thrown by attemptToChangeHeight and catch them in all other code paths

Hmm I don't quite follow. Preventing callers from needing to worry about `attemptChangeHeight` failure should be doable. Happy to chat about this.

Another confusing point is that `unlockHeight` only applies if `target == this.container_`.

Reply:
> Preventing callers from needing to worry about `attemptChangeHeight` failure should be doable.
It's doable but I'm not convinced that it is better.